### ISSUE TITLE
Exception when enabling a plugin - IllegalStateException: Cannot access delegate without a transaction

### ISSUE THREAD

**Issue by drlivsi** [aznamier/keycloak-event-listener-rabbitmq#52]
Keycloak versions:  23.0.1 (the same error in 21.1.2)

Exception when enabling plugin or changing User Event Settings or Admin Events Settings.

```
ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-1) Uncaught server error: java.lang.IllegalStateException: Cannot access delegate without a transaction
        at org.keycloak.models.cache.infinispan.RealmCacheSession.getRealmDelegate(RealmCacheSession.java:151)
        at org.keycloak.models.cache.infinispan.RealmCacheSession.getRealm(RealmCacheSession.java:410)
        at com.github.aznamier.keycloak.event.provider.RabbitMqConfig.calculateRoutingKey(RabbitMqConfig.java:44)
        at com.github.aznamier.keycloak.event.provider.RabbitMqEventListenerProvider.publishAdminEvent(RabbitMqEventListenerProvider.java:63)
        at org.keycloak.events.EventListenerTransaction.consumeAdminEventEntry(EventListenerTransaction.java:68)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at org.keycloak.events.EventListenerTransaction.commitImpl(EventListenerTransaction.java:60)
        at org.keycloak.models.AbstractKeycloakTransaction.commit(AbstractKeycloakTransaction.java:48)
        at org.keycloak.services.DefaultKeycloakTransactionManager.commit(DefaultKeycloakTransactionManager.java:146)
        at org.keycloak.services.DefaultKeycloakSession.closeTransactionManager(DefaultKeycloakSession.java:407)
        at org.keycloak.services.DefaultKeycloakSession.close(DefaultKeycloakSession.java:372)
        at org.keycloak.quarkus.runtime.transaction.TransactionalSessionHandler.close(TransactionalSessionHandler.java:55)
        at org.keycloak.quarkus.runtime.integration.jaxrs.CloseSessionHandler.closeSession(CloseSessionHandler.java:69)
        at org.keycloak.quarkus.runtime.integration.jaxrs.CloseSessionHandler.filter(CloseSessionHandler.java:65)
        at org.jboss.resteasy.reactive.server.handlers.ResourceResponseFilterHandler.handle(ResourceResponseFilterHandler.java:25)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:150)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:145)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:840)
```

**Comment by zluiten** [comment]
We experience the same.

Besides that, now when doing an updateRealm REST api call within a seconds after the realm had been created (and the GET realm api call is succesfully returning a result). The updateRealm REST api call will fail with a `500 Internal Server Error`. Not always though, it's like a fifty/fifty chance of failing.

No more information available at this moment though.

**Comment by huyqnguyen-it** [comment]
v3.0.4 add method get realm name instead of realm id.

use this to get realm name
String realmName = session.getContext().getRealm().getName();

**Comment by gilamarko** [comment]
Same error here, can we move forward with this?

**Comment by aznamier** [comment]
Thank you very much @huyqnguyen-it for the suggestion of how to fix it. Its is now included in 3.0.5.
